File: system/core/AxiomConstraintValidator.py
class AxiomConstraintValidator:
    """Ensures structural integrity and syntax compliance of the ACVD before GAX initiates semantic vetting at P2.

    Failure at this stage immediately triggers an IH protocol with a specific ACVD_STRUCTURE_MISS flag.
    This prevents complex GAX logic from attempting to parse malformed constraints.
    """
    def __init__(self, acvd_schema_path, trusted_schema):
        self.schema_path = acvd_schema_path
        self.trusted_schema = trusted_schema

    def load_and_validate(self) -> bool:
        try:
            # 1. Load the proposed ACVD structure (potential source of malformation)
            acvd_data = self._load_json(self.schema_path)
            # 2. Validate ACVD against the defined governance schema/contract
            self._validate_schema(acvd_data)
            # 3. Check internal constraint coherence (e.g., TEMM thresholds are numeric)
            self._check_coherence(acvd_data)
            return True
        except Exception as e:
            # Log detailed error to FSL and return failure flag
            IH_Sentinel.trigger_halt(f"ACVD_STRUCTURE_MISS: {e}")
            return False

    # Internal implementation methods for loading/schema validation omitted for brevity...

File: core/AxiomConstraintValidator.js
const fs = require('fs');
const path = require('path');
// Assuming error handling (IH_Sentinel equivalent) is consolidated in /utility (Objective 3)
const { triggerHalt } = require('../utility/ErrorHalt'); 

/**
 * AxiomConstraintValidator
 * Ensures structural integrity and syntax compliance of the ACVD (Axiom Constraint Validation Document)
 * before semantic vetting initiates. Failure triggers an IH protocol (ACVD_STRUCTURE_MISS).
 */
class AxiomConstraintValidator {
    /**
     * @param {string} acvdSchemaPath Path to the proposed ACVD structure.
     * @param {object} trustedSchema The governance schema contract.
     */
    constructor(acvdSchemaPath, trustedSchema) {
        this.schemaPath = acvdSchemaPath;
        this.trustedSchema = trustedSchema;
    }

    /**
     * Loads the ACVD and performs structural and coherence validation.
     * @returns {Promise<boolean>} True if validation succeeds, false otherwise.
     */
    async loadAndValidate() {
        try {
            // 1. Load the proposed ACVD structure
            const acvdData = await this._loadJson(this.schemaPath);
            
            // 2. Validate ACVD against the defined governance schema/contract
            this._validateSchema(acvdData);
            
            // 3. Check internal constraint coherence
            this._checkCoherence(acvdData);
            
            return true;
        } catch (e) {
            // Log detailed error and trigger halt
            triggerHalt(`ACVD_STRUCTURE_MISS: ${e.message}`);
            return false;
        }
    }

    /**
     * Internal: Loads JSON data from a file path.
     */
    async _loadJson(filePath) {
        if (!fs.existsSync(filePath)) {
            throw new Error(`ACVD file not found at ${filePath}`);
        }
        const rawData = fs.readFileSync(filePath, 'utf8');
        return JSON.parse(rawData);
    }

    /**
     * Internal: Validates the loaded data against the trusted schema. (Placeholder)
     */
    _validateSchema(data) {
        // Placeholder: Requires external schema validator (e.g., Ajv) based on this.trustedSchema.
    }

    /**
     * Internal: Checks internal coherence (e.g., data types, required fields). (Placeholder)
     */
    _checkCoherence(data) {
        // Placeholder: Requires specific knowledge of ACVD structure.
    }
}

module.exports = AxiomConstraintValidator;